README.md

```diff
@@ -2,31 +2,559 @@
+as much of a substring of "tomcat" as "tom".
+
+The `String` class has two methods to get a substring that will each create a
+new `String` object by copying part of an existing `String`:
+
+```java
+public class StringSubstringExample {
+    public static void main(String[] args) {
+        String greeting = "HELLO WORLD";
+        String substring1 = greeting.substring(6);  //characters at indices 6,7,8,9,10
+        String substring2 = greeting.substring(0, 5); //characters at indices 0,1,2,3,4
+        System.out.println(greeting);
+        System.out.println(substring1);
+        System.out.println(substring2);
+    }
+}
+```
+
+Recall that a string uses 0-based indexing, just like an array:
+
+![String with Indices](https://curriculum-content.s3.amazonaws.com/6676/java-mod2-strings/hello_world.png)
+
+The program prints:
+
+```text
+HELLO WORLD
+WORLD
+HELLO
+```
+
+Let's look a little closer at these two different `substring()` methods:
+
+
+- The first `substring()` method only takes in one parameter, the starting index
+  of the original `String`, `greeting`. It will then return the substring of the
+  original `String` beginning at the index given. So in this example,
+  `greeting.substring(6)` would evaluate to "World".
+- The second `substring()` method takes in two parameters, a starting index and
+  the ending index of the original `String`. This method returns a substring of
+  the original `String` starting at the starting index and ending at the
+  ending index. It should be noted that the ending index is **not** inclusive,
+  so `greeting.substring(0,5)` will return only "Hello" and not the space after
+  it.
+
+![string substring](https://curriculum-content.s3.amazonaws.com/6676/java-mod2-strings/string_substring.png)
+
+### Position of a substring
+
+| String Method                   | Description                                                                               |
+|---------------------------------|-------------------------------------------------------------------------------------------|
+| public int indexOf(String str)  | Returns the index within this string of the first occurrence of the specified substring.  |
+
+
+Sometimes we need to know the starting position of a substring.
+The `indexOf` method returns the starting index of a substring.
+Since string indexing in Java is 0-based, the method returns `-1`
+if the substring is not found.
+
+![hello there world](https://curriculum-content.s3.amazonaws.com/6676/java-mod2-strings/hello_there_world.png)
+
+```java
+String greeting = "HELLO THERE WORLD!";
+System.out.println(greeting.indexOf("THE")); //6 is the starting position of the substring "THE"
+System.out.println(greeting.indexOf("hi")); //-1 indicates "hi" is not found
+```
+
+By default, the `indexOf` method starts looking at the beginning of the string.
+The code below prints `1`, the position of the "E" in the word "hello".
```

Comment:
Maybe "hello" should be in all uppercase characters: "HELLO"

Reply:
fixed